databricks/provider.go

```diff
@@ -192,6 +199,12 @@ func providerConfigureAzureClient(d *schema.ResourceData, providerVersion string
 	} else if os.Getenv("DATABRICKS_AZURE_WORKSPACE_NAME") != "" {
 		tokenPayload.WorkspaceName = os.Getenv("DATABRICKS_AZURE_WORKSPACE_NAME")
 	}
+	// TODO: Can required field not be set?
+	if workspaceURL, ok := azureAuthMap["workspace_url"].(string); ok {
```

Comment:
terraform behaves weirdly because this azure_auth is not a set/list block but a map so env defaults variables do not propagate.

Reply:
Is there any reason it is type map over a list? Maybe we should refactor it into a list to avoid these issues

Reply:
yep we can refactor this to a list with a min items to 1... there should be no reason to not change it, only impact is that it may break existing provider configurations.